packages/app-builder-lib/src/node-module-collector/types.ts

```diff
@@ -11,6 +11,7 @@ export interface DependencyTree {
   readonly from?: string
   readonly workspaces?: string[]
   readonly path: string
+  __circularDependencyDetected?: boolean
   dependencies?: {
     [packageName: string]: DependencyTree
   }
```

Comment:
Side note: This doesn't need to be part of this PR fix. But we should really reconsider refactoring this implementation as right now, it's an interface of **all but 1** optional, which is hardly better than just using `any`

What are your thoughts that we can do there?

Reply:
In this interface, `name`, `version`, and `path` should all be set as non-optional. If any of them is empty, it will cause issues. I have already fixed that.
As for the refactoring you mentioned, I don't have a good idea at the moment.